### PR TITLE
Backport pyopenssl deprecation PR to 1.26.x

### PR DESCRIPTION
**Pull request by IvanLauLinTiong**
This PR is to backport the pyopenssl deprecation PR #2701 to branch 1.26.x

### CODE REVIEW

File: changelog/2691.removal.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Deprecated ``urllib3.contrib.pyopenssl`` module which will be removed in future urllib3 v2.x release.
```
Comment (sethmlarson) [inline changelog/2691.removal.rst:1]: Let's remove this file.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# This file is protected via CODEOWNERS
```
Comment (sethmlarson) [inline pyproject.toml:1]: Let's remove this file. The changes that were made to this file in https://github.com/urllib3/urllib3/pull/2701 should be done inside of `setup.py` and `setup.cfg`.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Thanks for opening this, here are two comments for you:

File: changelog/2691.removal.rst (hunk 1 above)
Comment (IvanLauLinTiong) [reply changelog/2691.removal.rst]: fixed in new commit

File: pyproject.toml (hunk 2 above)
Comment (IvanLauLinTiong) [reply pyproject.toml]: fixed in new commit

**Review by sethmlarson — DISMISSED** [review]
LGTM, thanks @IvanLauLinTiong!

**Comment by IvanLauLinTiong** [comment]
@sethmlarson , there 's a CI error:
image

For Python2.7, I think I have to change the `ModuleNotFoundError` to `ImportError` , isn't it?

**Comment by sethmlarson** [comment]
Yes, instead of doing any if condition about which exception to catch let's change the except clause to always be ImportError.

**Review by sethmlarson — APPROVED** [review]

**Comment by sethmlarson** [comment]
Awesome, thank you @IvanLauLinTiong! You can now claim the bounty for https://github.com/urllib3/urllib3/issues/2691 on Open Collective.

**Comment by IvanLauLinTiong** [comment]
thank you

**Pull request merged**